.circleci/config.yml

```diff
@@ -13,101 +13,122 @@ jobs:
+
+  build_dev_backend:
+    working_directory: /app/pixel
+    docker:
+      - image: candihub/pixel:$CIRCLE_BRANCH
+    steps:
+      - restore_cache:
+          keys:
+            - backend-dev-deps-v1-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+      - run:
+          name: Install development dependencies
+          command: |
+            make install-ci
       - save_cache:
           paths:
-            - ~/.local
-            - ~/pixel/.venv
-          key: backend-deps-v3-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+            - ~/.local/share/virtualenvs
+          key: backend-dev-deps-v1-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
 
   lint_backend:
-    working_directory: ~/pixel
+    working_directory: /app/pixel
     docker:
-      - image: circleci/python:3.6.1
-        environment:
-          PIPENV_VENV_IN_PROJECT: 1
+      - image: candihub/pixel:$CIRCLE_BRANCH
     steps:
-      - checkout
       - restore_cache:
           keys:
-          - backend-deps-v3-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
-      - run:
-          name: Add Pipenv & installed tools to the PATH
-          command: |
-            echo "export PATH=~/.local/bin:~/pixel/.venv/bin:$PATH" >> $BASH_ENV
+            - backend-dev-deps-v1-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
       - run:
           name: Lint the code
-          command: make lint
+          command: |
+            make lint-ci
 
   test_backend:
-    working_directory: ~/pixel
+    working_directory: /app/pixel
     docker:
-      - image: circleci/python:3.6.1
+      - image: candihub/pixel:$CIRCLE_BRANCH
+        environment:
+          DJANGO_SETTINGS_MODULE: pixel.settings
+          DJANGO_CONFIGURATION: Test
+          DJANGO_SECRET_KEY: ThisIsAnExampleKeyForDevPurposeOnly
+          POSTGRES_HOST: localhost
+          POSTGRES_DB: pixel
+          POSTGRES_USER: pixel
+          POSTGRES_PASSWORD: pixel42
+      - image: postgres:9.6
         environment:
-          PIPENV_VENV_IN_PROJECT: 1
+          POSTGRES_DB: pixel
+          POSTGRES_USER: pixel
+          POSTGRES_PASSWORD: pixel42
     steps:
-      - checkout
       - restore_cache:
           keys:
-          - backend-deps-v3-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
+            - backend-dev-deps-v1-{{ .Branch }}-{{ checksum "Pipfile.lock" }}
       - run:
-          name: Add Pipenv & installed tools to the PATH
+          name: Test the code
           command: |
-            echo "export PATH=~/.local/bin:~/pixel/.venv/bin:$PATH" >> $BASH_ENV
+            sleep 10
+            make test-ci
       - run:
-          name: Test the code
+          name: Create git repository
           command: |
-            make test
+            git init
+            git remote add origin $CIRCLE_REPOSITORY_URL
+            mkdir ~/.ssh
+            ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
+            git fetch
+            git reset origin/$CIRCLE_BRANCH
+            git checkout origin/$CIRCLE_BRANCH
```

Comment:
`coveralls` reporting needs a `git` repository to fully qualify its report and better integrate with GitHub. I tried to use an alternative coveralls client with `no-git` support and it works, but the coveralls job cannot be linked with our GitHub PR…

Reply:
But don't we get the link in the PR itself? In the different CI builds?

Reply:
Yes, we have this info in the environment, but python-coveralls does not support collecting info from the environment instead of `git`. Hence IMO we have only two solutions: 1. develop this feature in one of the two coveralls clients, 2. restore a git repository. From a pragmatic point-of-view, the second appeared the best to me.

Reply:
No I meant on GitHub:

![screenshot_20171025-121054](https://user-images.githubusercontent.com/217628/31993547-c9eaaea4-b97d-11e7-9518-b58d81289496.png)

Reply:
No, no webhook is called in this case.